base/record_replay.h

```diff
@@ -169,6 +170,67 @@ struct CompareMemberByPointerId {
   }
 };
 
+inline unsigned HashInt(uint32_t key) {
+  key += ~(key << 15);
+  key ^= (key >> 10);
+  key += (key << 3);
+  key ^= (key >> 6);
+  key += ~(key << 11);
+  key ^= (key >> 16);
+  return key;
+}
+
+inline unsigned HashInt(uint64_t key) {
+  key += ~(key << 32);
+  key ^= (key >> 22);
+  key += ~(key << 13);
+  key ^= (key >> 8);
+  key += (key << 3);
+  key ^= (key >> 15);
+  key += ~(key << 27);
+  key ^= (key >> 31);
+  return static_cast<unsigned>(key);
+}
+
+// Replay's hashing function for pointers, using the registered pointer id.
+template <typename T>
+struct ReplayPtrHash {
+  static unsigned GetHash(T* key) {
+    if (!recordreplay::IsRecordingOrReplaying("pointer-ids")) {
+      return HashInt((uint64_t)key);
+    }
+
+    int ptr = recordreplay::PointerId(key);
```

Comment:
The recordreplay:: usage isn't necessary here, since this is in the recordreplay namespace.